### PR TITLE
Adding dialogParentComm to test app

### PR DESCRIPTION
**Pull request by lakhveerk**
For more information about how to contribute to this repo, visit [this page](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md).

## Description

> Summarize the changes, including the goals and reasons for this change. Be sure to call out any technical or behavior changes that reviewers should be aware of.

Adding Dialog.url.dialogParentCommunication to test file.
Adding e2e test json files for dialog subcapabilites. 

related pr: https://github.com/OfficeDev/microsoft-teams-library-js/pull/2127

> If this Pull Request should close/resolve any issues when merged, use [the special syntax for that](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) here.

### Main changes in the PR:

1. <Change 1>
2. <Change 2>

### Validation performed:

1. <Step 1>
2. <Step 2>

### Unit Tests added:

> Unit tests are required for all changes. If no unit tests were added as part of this change, please explain why they aren't necessary.

<Yes/No>

### End-to-end tests added:

<Yes/No>

### Change file added:

> Ensure the change file meets the [formatting requirements](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md#change-log-using-beachball).

<Yes/No>

### Related PRs:

> Remove this section if n/a

### Next/remaining steps:

> List the next or remaining steps in implementing the overall feature in subsequent PRs (or is the feature 100% complete after this?).

> Remove this section if n/a

- [ ] Item 1
- [ ] Item 2

### Screenshots:

image

> Remove this section if n/a

| Before     | After      |
| ---------- | ---------- |
| < image1 > | < image2 > |

### CODE REVIEW

File: apps/teams-test-app/e2e-test-data/dialogCard.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "DialogCard",
```
Comment (KangxuanYe) [inline apps/teams-test-app/e2e-test-data/dialogCard.json:2]: Isn't it the sub capability of capability? If we are following the naming as page, i.e., pages.config, it needs to be Dialog Card. Same to others

File: apps/teams-test-app/e2e-test-data/dialogCard.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (KangxuanYe) [inline apps/teams-test-app/e2e-test-data/dialogCard.json:1]: It's reasonable to have those changes and separate each sub capability as a json file. But do you have a corresponding change in web hub sdk to retrieve those files and do test? Could you ping me for that? I would like to run it in local unless you told me that you have checked it in.

**Review by KangxuanYe — COMMENTED** [review]
Left some comments regarding of E2E test.

File: apps/teams-test-app/e2e-test-data/dialogCard.json (hunk 2 above)
Comment (lakhveerk) [reply apps/teams-test-app/e2e-test-data/dialogCard.json:1]: Don't I have to merge this PR in before I use it in host sdk? E2e test command does not seem to load data from my local teams-js. I will merge this and put a PR on host sdk side.

Comment (KangxuanYe) [reply apps/teams-test-app/e2e-test-data/dialogCard.json:1]: It seems like you are just editing UI so I would say YOU DON'T HAVE TO FOLLOW THE WAY AS BUILDING A NEW CAPABILITY, in the other words, yes, you will need this PR first and then make corresponding change in web host sdk.

**Review by KangxuanYe — APPROVED** [review]

**Review by shrshindeMSFT — APPROVED** [review]
LGTM 🚢

**Pull request merged**